source/fuzz/call_graph.cpp

```diff
@@ -20,12 +20,19 @@ namespace spvtools {
 namespace fuzz {
 
 CallGraph::CallGraph(opt::IRContext* context) {
-  // Initialize function in-degree and call graph edges to 0 and empty.
+  // Initialize function in-degree, call graph edges and corresponding maximum
+  // loop nesting depth to 0, empty and 0 respectively.
   for (auto& function : *context->module()) {
     function_in_degree_[function.result_id()] = 0;
     call_graph_edges_[function.result_id()] = std::set<uint32_t>();
+    function_max_loop_nesting_depth_[function.result_id()] = 0;
   }
 
+  // Record the maximum loop nesting depth for each edge, by keeping a map from
+  // pairs of function ids, where (A, B) represents a function call from A to B,
+  // to the corresponding maximum depth.
+  std::map<std::pair<uint32_t, uint32_t>, uint32_t> call_to_max_depth;
+
   // Consider every function.
   for (auto& function : *context->module()) {
```

Comment:
Up to you, but it might be nice to extract this whole loop into a private helper function with an appropriate name, as well as the code that comes after the call to ComputeTopologicalOrderOfFunctions.

That way the constructor would be fairly short and would consist mainly of calls to individually cohesive helpers.

Reply:
Done